sphinx/builders/htmlhelp.py

```diff
@@ -232,6 +239,36 @@ def update_page_context(self, pagename, templatename, ctx, event_arg):
         # type: (unicode, unicode, Dict, unicode) -> None
         ctx['encoding'] = self.encoding
 
+    def process_final_html(self, htm):
+        # type: (unicode) -> unicode
+        """If ``htmlhelp_ascii_output`` is True, non-ASCII characters
+           after <head> tag will be escaped to ASCII form.
+           e.g. ``ß`` -> ``&szlig;``.
+        """
+        # only process when htmlhelp_ascii_output is True
+        if not self.config.htmlhelp_ascii_output:
+            return htm
+
+        def process(string):
+            # type: (unicode) -> unicode
+            def escape(matchobj):
+                # type: (Match[unicode]) -> unicode
+                codepoint = ord(matchobj.group(0))
+                name = codepoint2name.get(codepoint)
+                if name is None:
+                    return '&#%d;' % codepoint
+                else:
+                    return '&%s;' % name
+            return re.sub(r'[^\x00-\x7F]', escape, string)
+
+        # escape non-ASCII characters after head tag
+        m = re.search(r'^.*?</head>', htm, re.DOTALL|re.IGNORECASE)
```

Comment:
I suppose `<head>` tag can be processed by htmlentities. It does not cause moji-bakes to CP1252 encodings.

Reply:
chm is from 20 years ago, a pre-Unicode age, it seems Microsoft doesn't develop it since long ago.
But it's a successful product, people still using it.
It would be nice to have a modern replacement for chm.